src/main/java/course_project/SeaBattle.java

```diff
@@ -0,0 +1,108 @@
+package course_project;
+
+import course_project.components.Coordinate;
+import course_project.components.PlayingField;
+import course_project.util.PlayingFieldPrinter;
+import course_project.util.user_input_reader.UserInputReader;
+import course_project.util.user_input_reader.UserShotInputReader;
+import course_project.util.ship_placer.ComputerShipPlacer;
+import course_project.util.ship_placer.UserShipPlacer;
+
+import java.util.Random;
+import java.util.Scanner;
+
+public class SeaBattle {
+
+    final PlayingField playersField = new PlayingField();
+    final PlayingField enemyField = new PlayingField();
+    final PlayingFieldPrinter printer = new PlayingFieldPrinter();
+    private final Scanner scanner = new Scanner(System.in);
+    private final Random randomGenerator = new Random();
+    private boolean isHit;
+
+    public void playVsComputer() {
+        placePlayersShips();
+        placeComputerShips();
+        makeMoves();
+        defineWinner();
+        closeInput();
```

Comment:
Opt: these methods should have return values. It helps to test them separately. Better not to use void methods.

Reply:
Opt: ship placement is difficult to understand (h1 hor on empty field does not work)